spec/tpm/key_attestation_spec.rb

```diff
@@ -7,69 +7,71 @@
+      hash_algorithm: hash_algorithm,
+      root_certificates: root_certificates
+    )
+  end
 
-    let(:certificates) { [certificate.to_der] }
+  let(:certificates) { [certificate.to_der] }
 
-    let(:certificate) do
-      create_certificate(attestation_key, root_certificate, root_key)
-    end
+  let(:certificate) do
+    create_certificate(attestation_key, root_certificate, root_key)
+  end
 
-    let(:root_certificates) { [root_certificate] }
+  let(:root_certificates) { [root_certificate] }
 
-    let(:root_certificate) { create_root_certificate(root_key) }
-    let(:root_key) { create_rsa_key }
+  let(:root_certificate) { create_root_certificate(root_key) }
+  let(:root_key) { create_rsa_key }
 
-    let(:certify_info) do
-      s_attest = TPM::SAttest.new
-      s_attest.magic = certify_info_magic
-      s_attest.attested_type = TPM::ST_ATTEST_CERTIFY
-      s_attest.extra_data.buffer = certify_info_extra_data
-      s_attest.attested.name.name.hash_alg = name_alg
-      s_attest.attested.name.name.digest = certify_info_attested_name_digest
+  let(:certify_info) do
+    s_attest = TPM::SAttest.new
+    s_attest.magic = certify_info_magic
+    s_attest.attested_type = TPM::ST_ATTEST_CERTIFY
+    s_attest.extra_data.buffer = certify_info_extra_data
+    s_attest.attested.name.name.hash_alg = name_alg
+    s_attest.attested.name.name.digest = certify_info_attested_name_digest
 
-      s_attest.to_binary_s
-    end
+    s_attest.to_binary_s
+  end
 
-    let(:signature_algorithm) { TPM::ALG_RSASSA }
-    let(:hash_algorithm) { TPM::ALG_SHA256 }
-    let(:signature) { attestation_key.sign(hash_function, to_be_signed) }
-
-    let(:certify_info_magic) { TPM::GENERATED_VALUE }
-    let(:certify_info_extra_data) { qualifying_data }
-    let(:certify_info_attested_name_digest) { OpenSSL::Digest::SHA1.digest(attested_object) }
-    let(:name_alg) { TPM::ALG_SHA1 }
-    let(:to_be_signed) { certify_info }
-
-    let(:attested_object) do
-      t_public = TPM::TPublic.new
-      t_public.alg_type = TPM::ALG_RSA
-      t_public.name_alg = name_alg
-      t_public.parameters.symmetric = TPM::ALG_NULL
-      t_public.parameters.scheme = TPM::ALG_RSASSA
-      t_public.parameters.key_bits = 1024
-      t_public.parameters.exponent = 0x00
-      t_public.unique.buffer = OpenSSL::PKey::RSA.new(1024).params["n"].to_s(2)
-
-      t_public.to_binary_s
-    end
+  let(:signature_algorithm) { TPM::ALG_RSASSA }
+  let(:hash_algorithm) { TPM::ALG_SHA256 }
+  let(:signature) { attestation_key.sign(hash_function, to_be_signed) }
+
+  let(:certify_info_magic) { TPM::GENERATED_VALUE }
+  let(:certify_info_extra_data) { qualifying_data }
+  let(:certify_info_attested_name_digest) { OpenSSL::Digest::SHA1.digest(public_area) }
+  let(:name_alg) { TPM::ALG_SHA1 }
+  let(:to_be_signed) { certify_info }
+
+  let(:attested_key) { create_rsa_key }
+  let(:attested_key_length) { attested_key.n.num_bits }
+  let(:public_area) do
```

Comment:
Is the change from `attested_object` to `public_area` relevant to this PR or just a variable name change? If it's just a name change I'd rather keep it unchanged or, if changed, go with the same name we are using in the documentation which is `certified_key` to avoid confusion.

Reply:
Its just a variable name change because it seemed strange to me to have an `attested_key` and `attested_object`, I had a hard time thinking about names here 😂